src/main/ruby/truffleruby/core/range.rb

```diff
@@ -344,7 +344,7 @@ def ===(value)
   end
 
   def inspect
-    result = "#{self.begin.inspect}#{exclude_end? ? "..." : ".."}#{Truffle::RangeOperations.endless?(self) ? "" : self.end.inspect}"
+    result = "#{(self.begin == nil && self.end != nil) ? "" : self.begin.inspect}#{exclude_end? ? "..." : ".."}#{(Truffle::RangeOperations.endless?(self) && self.begin != nil) ? "" : self.end.inspect}"
```

Comment:
We should probably have `RangeOperations.beginless?` for symmetry. Since it's a `nil` check, it should probably be `Primitive.range_endless/beginless` ultimately. Very minor & can be taken care of later.

Reply:
I can definitely add a `RangeOperations.beginless?` method, although if a range is beginless and endless `(nil..nil)` then in MRI both sides get printed, so the other end need to be checked anyways.

I personally would find it more visually appealing in this case to use neither `beginless?` nor `endless?` (unless the check on the other end happens in `beginless?` or `endless?`, but redefining `endless?` could break things).

Reply:
Good point, what do think of:

```ruby
def inspect
  sep = exclude_end? ? "..." : ".."
  if (Primitive.nil? self.begin && Primitive.nil? self.end)
    result = "nil#{sep}nil"
  else
    result = (Primitive.nil? self.begin ? "" : self.begin.inspect) + sep +
             (Primitive.nil? self.end ? "" : self.end.inspect)
  end
  Primitive.infect(result, self)
end
```

Reply:
I love it. 

I'll commit it to the PR and squash it with the original Range#inspect when I rearrange the the commits after I'm done?

On that note, is there a preferred organization? Such as: All specs first; or Range behaviour first, Array behaviour follows, etc; or specs precede relevant code changes; or it really doesn't matter.

Reply:
The order of commits doesn't matter for specs.